### ISSUE TITLE
Script editor keyboard shortcuts not working on Mac (increase/decrease font size, block comment)

### ISSUE THREAD

**Issue by GenevieveBuckley** [scijava/script-editor#20]
There's a subset of keyboard shortcuts for the Fiji script editor that I can't get working. I'm sure this is a non-exhaustive list, but the biggest annoyances are:
* Edit > Increase font size ⌘+ (this works if you select it from the menu, but not via keyboard shortcut)
* Edit > Decrease font size ⌘-  (this works if you select it from the menu, but not via keyboard shortcut)
* Commenting a code block with ⌘/

I **can** use these shortcuts with no trouble:
* ⌘R - run script
* ⌘A, ⌘X, ⌘C, ⌘P - select all, cut, copy, paste
* ⌘Z, ⌘Y - undo, redo
* ⌘F, ⌘G - find, go to line

*System details:*
Mac High Sierra, 10.13.5

Fiji version: 2.0.0-rc-67/1.52d
Build 1762a07c5c
Date: 2018-06-04T18:22:47+0000

$ java -version
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

**Comment by GenevieveBuckley** [comment]
@ctrueden Yes,, the `+` and `-` keys are to the right of `0` and the left of `delete`, and the `/` key is to the left of the right side shift key:

![english_notebook](https://user-images.githubusercontent.com/30920819/41945769-0f97a1b6-79f2-11e8-8c88-f38e402d2e1c.png)

[This list](https://support.apple.com/en-au/ht201794) doesn't specifically call out an 'Australian' layout, but it looks as though I have what it's calling an "English" layout (not "English - Great Britain" or "English - International").

**Comment by ctrueden** [comment]
Thanks @GenevieveBuckley, that layout is identical to the one on my MacBook Pro.

On my system, <kbd>⌘+</kbd> requires that <kbd>shift</kbd> also be pressed, since <kbd>+</kbd> is really just <kbd>shift</kbd>+</kbd>=</kbd>. The <kbd>⌘-</kbd> shortcut works for me in all cases. The <kbd>⌘/</kbd> shortcut only works if you have a language selected from the Language menu; otherwise, it does not know which comment style to use.

With the above comments in mind, given the high level of detail on your bug report, I'm going to proceed under the assumption that your system is different and indeed none of the three shortcuts are working for you whatsoever.

I tried creating an MCVE. Here is a Java class we can use as a playground:
```java
import java.awt.BorderLayout;
import java.awt.event.InputEvent;
import java.awt.event.KeyEvent;
import javax.swing.*;

public class ShortcutSandbox {
	public static void main(final String... args) { new ShortcutSandbox();	}

	private JMenu shortcuts;
	private JLabel text;

	public ShortcutSandbox() {
		final int modifierCode = InputEvent.META_DOWN_MASK;
		System.setProperty("apple.laf.useScreenMenuBar", "true");

		JFrame frame = new JFrame("Keyboard Shortcut Test");
		frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
		JPanel pane = new JPanel();
		pane.setLayout(new BorderLayout());
		text = new JLabel("[Try some menu entries]");
		pane.add(text, BorderLayout.CENTER);
		frame.setContentPane(pane);

		JMenuBar menubar = new JMenuBar();
		frame.setJMenuBar(menubar);
		shortcuts = new JMenu("Shortcuts");
		menubar.add(shortcuts);

		createShortcut("Tee", KeyStroke.getKeyStroke(KeyEvent.VK_T, modifierCode));
		createShortcut("Zero", KeyStroke.getKeyStroke(KeyEvent.VK_0, modifierCode));
		createShortcut("Plus", KeyStroke.getKeyStroke(KeyEvent.VK_PLUS, modifierCode));
		createShortcut("Minus", KeyStroke.getKeyStroke(KeyEvent.VK_MINUS, modifierCode));
		createShortcut("Slash", KeyStroke.getKeyStroke(KeyEvent.VK_SLASH, modifierCode));

		createShortcut("Shift+Equals",
			KeyStroke.getKeyStroke(KeyEvent.VK_EQUALS, modifierCode | InputEvent.SHIFT_DOWN_MASK));

		frame.pack();
		frame.setVisible(true);
	}

	private JMenuItem createShortcut(String label, KeyStroke accelerator) {
		JMenuItem item = new JMenuItem(label);
		item.setAccelerator(accelerator);
		shortcuts.add(item);
		item.addActionListener(evt -> text.setText(label));
		return item;
	}
}
```
It should be possible to compile and run this on the command line with:
```
javac ShortcutSandbox.java
java ShortcutSandbox
```

On my system, all keyboard shortcuts except for <kbd>⌘+</kbd> work. (If you comment out the "Shift+Equals" shortcut, then <kbd>⌘+</kbd> does still flash the menu when <kbd>shift</kbd> is included in the combo, but does not trigger the corresponding action.)

In the script-editor source code in `TextEditor.java`, the `KeyStroke` objects are constructed using `KeyStroke.getKeyStroke(int, int)` which is what the code above uses too; the ["Increase font size" code](https://github.com/scijava/script-editor/blob/script-editor-0.2.0/src/main/java/org/scijava/ui/swing/script/TextEditor.java#L293-L294) uses `KeyEvent.VK_PLUS`, which [I guess we should change to `KeyEvent.VK_EQUALS` with a shift modifier](https://stackoverflow.com/q/15605109/1207769). And really, all of these special characters work differently on different regional layouts, which makes coding it to work on everyone's machine quite tricky.

I don't have more time to dig on this right now, but please ping back here if you uncover any further information, workarounds, thoughts, etc.

**Comment by GenevieveBuckley** [comment]
> With the above comments in mind, given the high level of detail on your bug report, I'm going to proceed under the assumption that your system is different and indeed none of the three shortcuts are working for you whatsoever.

This is too generous :) Turns out this is true only for some shortcuts, and only for a specific set of circumstances.

### The <kbd>⌘/</kbd> shortcut
<kbd>⌘/</kbd> does not work for me if the language selected is:
1. IJ1 Macro, or
2. R

It does work well for Beanshell, Clojure, Groovy, Java, JavaScript, Ruby, and Scala.

### The <kbd>⌘+</kbd> and <kbd>⌘-</kbd> shortcuts
> On my system, ⌘+ requires that shift also be pressed, since + is really just shift+=. 

This totally fooled me - since the terminal, TextEdit, Atom, etc. also have <kbd>⌘+</kbd> listed in their menus as the shortcut for increasing text size (just like the script editor), and do not require you to press shift... I didn't even consider that.

<kbd>⌘</kbd>+<kbd>+</kbd> does not increase font size, but
<kbd>⌘</kbd>+<kbd>Shift</kbd>+<kbd>+</kbd> does work. Thanks! 

>  The ⌘- shortcut works for me in all cases.

Me too.
<kbd>⌘-</kbd> does decrease font size, and
<kbd>⌘</kbd>+<kbd>Shift</kbd>+<kbd>-</kbd> also does decrease font size.  Since <kbd>Shift</kbd>+<kbd>-</kbd> technically produces an underscore it's a little odd that this works, given the explanation above about the difference between <kbd>⌘=</kbd> / <kbd>⌘+</kbd>. I would have expected symmetric behaviour.

### ShortcutSandbox playground
From the keyboard:
<kbd>⌘+</kbd> does not register any response.
<kbd>⇧⌘-</kbd> is not listed in the sandbox menu, but when you enter this on the keyboard it registers as 'Minus'.

<kbd>⌘T</kbd> registers 'Tee', as expected.
<kbd>⌘0</kbd> registers 'Zero', as expected.
<kbd>⌘-</kbd> registers 'Minus', as expected.
<kbd>⌘/</kbd> registers 'Slash', as expected.
<kbd>⇧⌘+</kbd> registers 'Shift+Equals', as expected.

> I don't have more time to dig on this right now, but please ping back here if you uncover any further information, workarounds, thoughts, etc.

This is completely fine, I'd say it's a fairly low priority. 
Font sizes are adjustable with <kbd>⌘-</kbd>and <kbd>⇧⌘+</kbd>, and if I happen to want a comment block in an IJ1 Macro (ie: rarely) I can toggle the language to Java briefly. I'm ok with that as a workaround.
